`ChaCha20With64BitNonce.stream` uses wrong type for `counter` parameter

### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

Try to compile code that uses `std.crypto.stream.chacha.ChaCha20With64BitNonce.stream`:
1. `counter` parameter is `u32` not `u64`
2. As a result, you also get a compile error from std:
```
/usr/lib/zig/std/crypto/chacha20.zig:607:50: error: type 'u5' cannot represent integer value '32'
            c[1] = @as(u32, @truncate(counter >> 32));
                                                 ^~
``` 

### Expected Behavior

`ChaCha20With64BitNonce.stream` should have `counter: u64` parameter like `.xor` does

--- a/lib/std/crypto/chacha20.zig
+++ b/lib/std/crypto/chacha20.zig
@@ -590,21 +590,21 @@ fn ChaChaWith64BitNonce(comptime rounds_nb: usize) type {
 
             const k = keyToWords(key);
             var c: [4]u32 = undefined;
-            c[0] = @as(u32, @truncate(counter));
-            c[1] = @as(u32, @truncate(counter >> 32));
+            c[0] = @truncate(counter);
+            c[1] = @truncate(counter >> 32);
             c[2] = mem.readInt(u32, nonce[0..4], .little);
             c[3] = mem.readInt(u32, nonce[4..8], .little);
             ChaChaImpl(rounds_nb).chacha20Xor(out, in, k, c, true);
         }
 
         /// Write the output of the ChaCha20 stream cipher into `out`.
-        pub fn stream(out: []u8, counter: u32, key: [key_length]u8, nonce: [nonce_length]u8) void {
+        pub fn stream(out: []u8, counter: u64, key: [key_length]u8, nonce: [nonce_length]u8) void {
             assert(out.len <= 64 * (@as(u71, 1 << 64) - counter));
 
             const k = keyToWords(key);
             var c: [4]u32 = undefined;
-            c[0] = @as(u32, @truncate(counter));
-            c[1] = @as(u32, @truncate(counter >> 32));
+            c[0] = @truncate(counter);
+            c[1] = @truncate(counter >> 32);
             c[2] = mem.readInt(u32, nonce[0..4], .little);
             c[3] = mem.readInt(u32, nonce[4..8], .little);
             ChaChaImpl(rounds_nb).chacha20Stream(out, k, c, true);

